src/test/java/io/stargate/sgv2/jsonapi/service/operation/OperationAttemptAssertions.java

```diff
@@ -0,0 +1,199 @@
+            })
+        .when(target)
+        .executeStatement(any());
+    return fixture;
+  }
+
+  public FixtureT assertStatus(OperationAttempt.OperationStatus status, String message) {
+    assertThat(target.status())
+        .as("Status should be %s when: %s", status, message)
+        .isEqualTo(status);
+    return fixture;
+  }
+
+  public FixtureT assertFailure(Class<? extends Throwable> clazz, String message) {
+    assertThat(target.failure())
+        .as("Attempt should have the failure class %s when: %s", clazz.getSimpleName(), message)
+        .isPresent()
+        .get()
+        .isInstanceOf(clazz);
+    return fixture;
+  }
+
+  public FixtureT assertFailure(Throwable throwable, String message) {
+    assertThat(target.failure())
+        .as("Attempt should have the failure %s when: %s", throwable.toString(), message)
+        .isPresent()
+        .get()
+        .isSameAs(throwable);
+    return fixture;
+  }
+
+  public FixtureT assertFailureEmpty(String message) {
+    assertThat(target.failure()).as("Attempt failure should be empty when: %s", message).isEmpty();
+    return fixture;
+  }
+
+  public FixtureT verifyExecuteStatementCalled(int times, String message) {
+    verify(
+            target,
+            times(times)
+                .description("execute() called %s times when: %s".formatted(times, message)))
+        .executeStatement(any());
+    return fixture;
+  }
+
+  public FixtureT verifyOnCompletionCalled(int times, String message) {
+
+    verify(
+            target,
+            times(times)
+                .description("onCompletion() called %s times when: %s".formatted(times, message)))
+        .onCompletion(any(), any(), any());
+    return fixture;
+  }
+
+  public FixtureT verifyOnCompletionResultSet(AsyncResultSet expectedResultSet, String message) {
+    verify(
+            target,
+            times(1)
+                .description(
+                    "onCompletion() called 1 time with resultset %s when: %s"
+                        .formatted(Objects.toString(expectedResultSet, "NULL"), message)))
+        .onCompletion(any(), eq(expectedResultSet), any());
+    return fixture;
+  }
+
+  public FixtureT verifyOnCompletionThrowable(Throwable expectedThrowable, String message) {
+    verify(
+            target,
+            times(1)
+                .description(
+                    "onCompletion() called 1 time with throwable %s when: %s"
+                        .formatted(Objects.toString(expectedThrowable, "NULL"), message)))
+        .onCompletion(any(), any(), eq(expectedThrowable));
+    return fixture;
+  }
+
+  public FixtureT verifyOnSuccessCalled(int times, String message) {
+    verify(
+            target,
+            times(times)
+                .description("onSuccess() called %s times when: %s".formatted(times, message)))
+        .onSuccess(any());
+    return fixture;
+  }
+
+  public FixtureT verifyOnSuccessResultSet(AsyncResultSet expected, String message) {
```

Comment:
I saw you also check these times(x) method in multiple place.

Take OnSuccess as example.
When you do the fluent way of test in ReadAttemptTest.

You have this:

```
 .verifyOnSuccessCalled(1, msg)
 .attempt()
 .verifyOnSuccessResultSet(fixture.resultSet(), msg)
 .attempt()
```

verifyOnSuccessResultSet() is doing the amount of method call check again.
Wonder if these two can be combined?